### ISSUE TITLE
Enums as names

### ISSUE THREAD

**Issue by USSX-Hares** [lidatong/dataclasses-json#101]
Hi.

Is there any option to encode enums by their name not their value?
I.e., is there any options to make this code:
```python
from enum import Enum, auto
from dataclasses import dataclass
from dataclasses_json import DataClassJsonMixin

class Status(Enum):
    Alive = auto()
    Dead = auto()
    Comatose = auto()

@dataclass
class Human(DataClassJsonMixin):
    name: str
    status: Status = Status.Alive

if (__name__ == '__main__'):
    santa = Human("John")
    barbara = Human("Joan", Status.Comatose)
    forever = Human("Helen", Status.Dead)
    
    print(santa.to_json())
    print(barbara.to_json())
    print(forever.to_json())
```

... instead of printing:
```json
{"name": "John", "status": 1}
{"name": "Joan", "status": 3}
{"name": "Helen", "status": 2}
```

... print:
```json
{"name": "John", "status": "Alive"}
{"name": "Joan", "status": "Comatose"}
{"name": "Helen", "status": "Dead"}
```

And do so for the `Human.from_json()` as well.
For me, it looks like additional encoder/decoder flag, something like `enums_by_id: bool = False`
(default should be False to save the remain backwards-compatible)

Thank you!

P.S.
Could try to implement this later.

**Comment by USSX-Hares** [comment]
Of course, it is always possible to do something like this:
```python
class Status(Enum):
    Alive = 'Alive'
    Dead = 'Dead'
    Comatose = 'Comatose'
```

or even
```python
def str_decorator_on_enum(cls):
    ...

@str_decorator_on_enum
class Status(Enum):
    Alive = auto()
    Dead = auto()
    Comatose = auto()
```

But this could require some code changes on the already existing classes and thus could lead to breaking existing functionality.

**Comment by nathan5280** [comment]
I had this as an example I did to handle enums by name.

class Sex(Enum):
    female = 0
    male = 1


class SexField(fields.Field):
    def _serialize(self, value, attr, obj, **kwargs):
        return value.name

    def _deserialize(self, value, attr, data, **kwargs):
        return Sex[value]


sex_field = {
    "dataclasses_json": {
        "encoder": lambda sex: sex.name,
        "decoder": lambda name: Sex(name),
        "mm_field": SexField(),
    }
}


@dataclass_json
@dataclass
class Actor:
    character_name: str
    actor_name: str
    sex: Sex = field(metadata=sex_field)

**Comment by USSX-Hares** [comment]
@nathan5280 hmmm...
probably could make a simple decorator for that...

**Comment by Peilonrayz** [comment]
@nathan5280 That code looks a lot like #61.

I don't think using a decorator would be the best idea for the general case, like in #61.

I've not looked at `mm.py` for a while, but passing those options to the fields should be easy from recollection.

**Comment by USSX-Hares** [comment]
```python
from enum import Enum, auto
from typing import TypeVar, Type
from dataclasses import dataclass, field
from dataclasses_json import DataClassJsonMixin
from marshmallow import fields

T = TypeVar('T', bound=Enum)
def string_enum(cls: Type[T]) -> Type[T]:
    class EnumField(fields.Field):
        def _serialize(self, value, attr, obj, **kwargs):
            return value.name
        
        def _deserialize(self, value, attr, data, **kwargs):
            return cls[value]
    
    if (not hasattr(cls, '__metadata__')):
        setattr(cls, '__metadata__', dict())
    
    metadata = {
        "dataclasses_json": {
            "encoder": lambda v: v.name,
            "decoder": lambda name: cls[name],
            "mm_field": EnumField(),
        }
    }
    
    cls.__metadata__.update(metadata)
    return cls

@string_enum
class Status(Enum):
    Alive = auto()
    Dead = auto()
    Comatose = auto()

@dataclass
class Human(DataClassJsonMixin):
    name: str
    status: Status = field(default=Status.Alive, metadata=Status.__metadata__)

if (__name__ == '__main__'):
    santa = Human("John")
    barbara = Human("Joan", Status.Comatose)
    forever = Human("Helen", Status.Dead)
    
    print("Dumping...")
    print(santa.to_json())
    print(barbara.to_json())
    print(forever.to_json())
    
    print("Loading...")
    print(Human.from_json('{"name": "John", "status": "Alive"}'))
    print(Human.from_json('{"name": "Joan", "status": "Comatose"}'))
    print(Human.from_json('{"name": "Helen", "status": "Dead"}'))    
```

**Comment by lidatong** [comment]
Thanks and yes I think this is a useful feature, I will probably address after #61

**Comment by ysangkok** [comment]
would be nice to have it work with [marshmallow_enum.EnumField](https://pypi.org/project/marshmallow-enum/)

**Comment by George3d6** [comment]
Any updates on this one?

**Comment by thoroc** [comment]
@George3d6 would that closed issue help https://github.com/lidatong/dataclasses-json/issues/240 ?

**Comment by erikfinnman** [comment]
I got this to work using the `StrEnum` type that was introduced in Python 3.11: https://docs.python.org/3/library/enum.html#enum.StrEnum.
